### PR TITLE
Fix ReJIT failures

### PR DESCRIPTION
**Pull request by davmason**

### CODE REVIEW

**Comment by davmason** [comment]
@jkotas @noahfalk What is the guidance for when a thread can either be called from a non-runtime thread or called on a runtime thread but needs to be in coop mode? I see there is `GCX_COOP_NO_THREAD_BROKEN` but the name implies I shouldn't use it. There is `GCX_MAYBE_COOP` but it asserts that the thread exists even if the condition is false. I'm also not sure what the practical difference would be between `GCX_COOP_NO_THREAD_BROKEN()` and `GCX_MAYBE_COOPY(GetThread() != NULL)`. They seem to be functionally the same.

**Comment by jkotas** [comment]
`GCX_COOP_NO_THREAD_BROKEN` does not work in all cases as the name suggests. We get away with it in the few cases where it is used today because of there are other invariants in the system (the broken case always happens when the runtime is suspended for GC) that make it ok for those few cases. It do not see any guarantee like this for your use here.

If you need to switch to cooperative mode on a thread that the runtime have not see before, you need to call `SetupThread` on it in the profiler entrypoint.

**Comment by davmason** [comment]
@noahfalk I know you're busy today with code reviews, if you have time tonight to look at this I would like to get it in for preview 7. It's not the end of the world if not though. The code that is up for review right now shouldn't change, I ran the profiler tests and it fixes all issues but one and I think that is a test issue.

File: src/vm/proftoeeinterfaceimpl.inl [hunk 1]
```diff
@@ -63,9 +63,10 @@ inline BOOL AreCallbackStateFlagsSet(DWORD dwFlags)
     BOOL fRet;
     BEGIN_GETTHREAD_ALLOWED_IN_NO_THROW_REGION;
     DWORD dwProfilerCallbackFullStateFlags = pThread->GetProfilerCallbackFullState();
-    if ((dwProfilerCallbackFullStateFlags & COR_PRF_CALLBACKSTATE_FORCEGC_WAS_CALLED) != 0)
+    if (((dwProfilerCallbackFullStateFlags & COR_PRF_CALLBACKSTATE_FORCEGC_WAS_CALLED) != 0)
+        || ((dwProfilerCallbackFullStateFlags & COR_PRF_CALLBACKSTATE_REJIT_WAS_CALLED) != 0))
```
Comment (noahfalk) [inline src/vm/proftoeeinterfaceimpl.inl]: Not necessarily a problem but a bit worrisome, if a thread had already called into managed code and then it called ReJIT anything we will permanently relax the checks for it.

**Comment by davmason** [comment]
Thanks Noah!

**Pull request merged**